lib/ex_force/oauth.ex

```diff
@@ -96,12 +96,35 @@ defmodule ExForce.OAuth do
     refresh_token: "refresh_token"
   )
   ```
+
+  ### `jwt_token`
+
+    ```elixir
+  ExForce.OAuth.get_token(
+    "https://login.salesforce.com",
+    grant_type: "jwt",
+    username: "username",
+    client_id: "client_id",
+    jwt_key: "jwt_key"
+  )
+  ```
+
   """
 
   @spec get_token(ExForce.Client.t() | String.t(), list) ::
           {:ok, OAuthResponse.t()} | {:error, :invalid_signature | term}
 
-  def get_token(url, payload) when is_binary(url), do: url |> build_client() |> get_token(payload)
+  def get_token(url, payload) when is_binary(url) do
+    url
+    |> build_client
+    |> then(
+      &if Keyword.fetch!(payload, :grant_type) == "jwt" do
+        get_token_jwt(&1, url, payload)
+      else
+        get_token(&1, payload)
+      end
+    )
+  end
```

Comment:
I created a version of `get_token` for jwt, named `get_token_jwt` because there are several differences in the jwt process vs the currently supported flows (for example, there is no signature verification for the jwt flow), and I thought a new function was cleaner than trying to update the existing function to handle the jwt process.  So I am using the wrapper function to branch and call either the existing `get_token` or the new 
`get_token_jwt`, based on the specified `grant_type`.  Plus the jwt process requires the url, which the wrapper function has available to it but `get_token` does not.

Two questions for you:
1. Do you want me to put the new logic in the existing `get_token(client, payload)` function rather than create the new function `get_token_jwt`?  
2. If yes to (1), how would you suggest I access the url in `get_token(client, payload)`?  Can I extract it from the passed in Tesla client, `client`?  I briefly looked at the Tesla docs and did not see how to do that.  Or should I add a url parameter to `get_token(client,payload)`, yielding `get_token(client, url, payload)`, and pass it in from the wrapper?
 
I am happy to update my changes if you can give me a bit of direction on the questions above. Thank you!

Reply:
I see now how to extract the url from `client`.  I will try and refactor.

Reply:
OK - Let me take a crack at this.  I think I understand your direction here.

Reply:
OK - I refactored this to try and follow your direction for the `get_token` method.  I am a bit concerned now that `verify_signature` is doing two jobs - verifying the signature and formatting the output into an %OAuthResponse.  In the case of the jwt path, it is actually _only_ formatting and not verifying anything at all.  I wonder if we should separate the formatting into another function?

Reply:
Any thoughts on the changes I have made here?  Hopefully you will find them to be aligned with your suggestions.